### PR TITLE
update micromatch and add braces

### PR DESCRIPTION
**Pull request by vemoo**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/micromatch/micromatch/blob/master/README.md
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/braces/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "max-line-length": false
```
Comment (plantain-00) [inline types/braces/tslint.json]: For new packages, this file should only `"extends": "dtslint/dt.json"`.
If you have to disable it, you can disable the rule by disabling it in the source code.

File: types/micromatch/index.d.ts [hunk 2]
```diff
@@ -1,171 +1,621 @@
-// Type definitions for micromatch 2.3.7
-// Project: https://github.com/jonschlinkert/micromatch
+// Type definitions for micromatch 3.1
```
Comment (plantain-00) [inline types/micromatch/index.d.ts]: If you want to update the package to a new major version, you should follow the guide: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#i-want-to-update-a-package-to-a-new-major-version

File: types/micromatch/tslint.json [hunk 3]
```diff
@@ -1,79 +1,6 @@
...
-        "space-before-function-paren": false,
-        "space-within-parens": false,
-        "strict-export-declare-modifiers": false,
-        "trim-file": false,
-        "triple-equals": false,
-        "typedef-whitespace": false,
-        "unified-signatures": false,
-        "void-return": false,
-        "whitespace": false
+        "max-line-length": false
```
Comment (plantain-00) [inline types/micromatch/tslint.json]: Same here

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/micromatch/index.d.ts (hunk 2 above)
Comment (vemoo) [reply types/micromatch/index.d.ts]: How should I go about restoring the previous version, should I edit the commit, or just add the original files in a new commit?

Comment (plantain-00) [reply types/micromatch/index.d.ts]: I think adding them as a new commit is good to go.

**Review by plantain-00 — APPROVED** [review]

**Pull request merged**